rust/kernel/thread.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::bindings;
+use crate::c_types;
+
+use alloc::boxed::Box;
+use core::ops::FnOnce;
+
+#[no_mangle]
+unsafe extern "C" fn rust_thread_func(data: *mut c_types::c_void) -> c_types::c_int {
+    Box::from_raw(data as *mut Box<dyn FnOnce()>)();
+    0
+}
+
+pub struct Thread {
+    task: *mut bindings::task_struct,
+}
+
+impl Thread {
+    pub fn new<F>(f: F, name: &str) -> Self
+    where
+        F: FnOnce(),
+        F: Send + 'static,
+    {
+        let bf: Box<dyn FnOnce() + 'static> = Box::new(f);
+
+        unsafe {
+            let task = bindings::kthread_create_on_node(
+                Some(rust_thread_func),
+                Box::into_raw(Box::new(bf)) as *mut _,
```

Comment:
Double boxing? Is this because of the fat pointer? Can we avoid boxing altogether? (See my previous comment.)

In any event, you need to use `Box::try_new`.

Reply:
Yeah, the double boxing is because of the fat pointer.

Reply:
Please add a comment on the double boxing.